cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiTest.java

```diff
@@ -290,4 +290,28 @@ public void testChangeLoggers(VertxTestContext context) throws InterruptedExcept
                             async.flag();
                         }))));
     }
-}
+
+    @IsolatedTest
+    public void testHierarchy() {
+        String rootLevel = "TRACE";
+        String desired = "log4j.rootLogger=" + rootLevel + ", CONSOLE\n" +
+                "log4j.logger.oorg.apache.zookeeper=WARN\n" +
+                "log4j.logger.oorg.I0Itec.zkclient=INFO\n" +
+                "log4j.logger.oorg.reflections.Reflection=INFO\n" +
+                "log4j.logger.oorg.reflections=FATAL\n" +
+                "log4j.logger.foo=WARN\n" +
+                "log4j.logger.foo.bar=TRACE\n" +
+                "log4j.logger.oorg.eclipse.jetty.util=DEBUG\n" +
+                "log4j.logger.foo.bar.quux=DEBUG";
+
+        KafkaConnectApiImpl client = new KafkaConnectApiImpl(vertx);
+        OrderedProperties ops = new OrderedProperties();
+        ops.addStringPairs(desired);
+        assertEquals(client.getLevel("foo.bar", ops.asMap()), "TRACE");
```

Comment:
The expect value should come first., so you need to flip the arguments on each assert.